internal/pkg/manifest/validate.go

```diff
@@ -0,0 +1,410 @@
+func (*HTTPHealthCheckArgs) Validate() error {
+	return nil
+}
+
+// Validate returns if Alias is configured correctly.
+func (*Alias) Validate() error {
+	return nil
+}
+
+// Validate returns if TaskConfig is configured correctly.
+func (t *TaskConfig) Validate() error {
+	var err error
+	if err = t.Platform.Validate(); err != nil {
+		return fmt.Errorf(`validate "platform": %w`, err)
+	}
+	if err = t.Count.Validate(); err != nil {
+		return fmt.Errorf(`validate "count": %w`, err)
+	}
+	if err = t.ExecuteCommand.Validate(); err != nil {
+		return fmt.Errorf(`validate "exec": %w`, err)
+	}
+	if err = t.Storage.Validate(); err != nil {
+		return fmt.Errorf(`validate "storage": %w`, err)
+	}
+	return nil
+}
+
+// Validate returns if PlatformArgsOrString is configured correctly.
+func (p *PlatformArgsOrString) Validate() error {
+	if !p.PlatformArgs.isEmpty() {
+		return p.PlatformArgs.Validate()
+	}
+	return nil
+}
+
+// Validate returns if PlatformArgsOrString is configured correctly.
+// TODO: add validation once "feat/pencere" is merged.
+func (*PlatformArgs) Validate() error {
+	return nil
+}
+
+// Validate returns if Count is configured correctly.
+func (c *Count) Validate() error {
+	if !c.AdvancedCount.IsEmpty() {
+		return c.AdvancedCount.Validate()
+	}
+	return nil
+}
+
+// Validate returns if AdvancedCount is configured correctly.
+func (a *AdvancedCount) Validate() error {
+	if a.Spot != nil && a.hasAutoscaling() {
+		return &errFieldMutualExclusive{
+			firstField:  "spot",
+			secondField: "range/cpu_percentage/memory_percentage/requests/response_time",
+		}
+	}
+	if !a.Range.IsEmpty() {
+		if err := a.Range.Validate(); err != nil {
+			return fmt.Errorf(`validate "range": %w`, err)
+		}
+	}
+	if a.CPU != nil || a.Memory != nil || a.Requests != nil || a.ResponseTime != nil {
+		return &errFieldMustBeSpecified{
+			missingField:      "range",
+			conditionalFields: []string{"cpu_percentage", "memory_percentage", "requests", "response_time"},
+		}
+	}
+	return nil
+}
+
+// Validate returns if Range is configured correctly.
+func (r *Range) Validate() error {
+	if !r.RangeConfig.IsEmpty() {
+		return r.RangeConfig.Validate()
+	}
+	return r.Value.Validate()
+}
+
+// Validate returns if IntRangeBand is configured correctly.
+func (r *IntRangeBand) Validate() error {
+	str := string(*r)
+	minMax := intRangeBandRegexp.FindStringSubmatch(str)
+	// Valid minMax example: ["1-2", "1", "2"]
+	if len(minMax) != 3 {
+		return fmt.Errorf("invalid range value %s. Should be in format of ${min}-${max}", str)
+	}
+	// Guaranteed by intRangeBandRegexp.
+	min, err := strconv.Atoi(minMax[1])
+	if err != nil {
+		return err
+	}
+	max, err := strconv.Atoi(minMax[2])
+	if err != nil {
+		return err
+	}
+	if min <= max {
+		return nil
+	}
+	return &errMinGreaterThanMax{
+		min: min,
+		max: max,
+	}
+}
+
+// Validate returns if RangeConfig is configured correctly.
+func (r *RangeConfig) Validate() error {
+	min, max := aws.IntValue(r.Min), aws.IntValue(r.Max)
```

Comment:
Should we also make sure that if `min` is specified, then `max` must be specified too?